cli/exec/exec.go

```diff
@@ -146,13 +146,17 @@ func execWithAxis(ctx context.Context, c *cli.Command, file, repoPath string, ax
 	}
 
 	environ := metadata.Environ()
-	var secrets []compiler.Secret
-	for key, val := range metadata.Workflow.Matrix {
-		environ[key] = val
-		secrets = append(secrets, compiler.Secret{
-			Name:  key,
-			Value: val,
-		})
+
+	secretMap := metadata.Workflow.Matrix
+	secretQuerier := func(name string) (*compiler.Secret, error) {
+		secret, found := secretMap[strings.ToLower(name)]
```

Comment:
currently we use case-insensitive for name matiching ... If ok I want to make a breaking change afterwards that make it truly case sensitive ...

Reply:
PS: I would do that in an dedicated pull